app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+
+    if user&.authenticate params[:session][:password]
```

Comment:
toán tử safe navigation `&.` có tác dụng gì?

Reply:
safe navigation dùng để rút gọn toán tử &&. Thay vì viết user && user.authenticate ta có thể viết gọn lại thành user&.authenticate

Reply:
tìm hiểu lại nhé, chưa chuẩn đâu em nhá
xe toán tử này với method `try` có gì khác nhau

Reply:
cú pháp &. chỉ hoạt động khi object nil còn trường hợp false thì bắn ra exception. Method try luôn trả về nill. Việc sử dụng &. giúp code ngắn gọn hơn try(...).

Reply:
@HNInternRubyNguyenBaoLongB Chưa đúng đâu em. Em đang hiểu sai về hàm try rồi